nexus/src/app/background/sync_switch_configuration.rs

```diff
@@ -587,6 +590,154 @@ impl BackgroundTask for SwitchPortSettingsManager {
                             bgp_announce_prefixes.insert(bgp_config.bgp_announce_set_id, prefixes);
                         }
 
+                        let ttl: Option<u8> = match peer.min_ttl.as_ref() {
+                            Some(x) => {
+                                match x.0.try_into() {
+                                    Ok(ttl) => Some(ttl),
+                                    Err(e) => {
+                                        error!(log,
+                                            "peer ttl out of range {} > 255",
+                                            x;
+                                            "error" => ?e
+                                        );
+                                        return json!({
+                                            "error":
+                                                format!(
+                                                    "peer ttl out of range: \
+                                                        {:#}",
+                                                    e
+                                                )
+                                        });
+                                    }
+                                }
+                            }
+                            None => None,
+                        };
+
+                        let communities = match self.datastore.communities_for_peer(
+                            opctx,
+                            peer.port_settings_id,
+                            &peer.interface_name,
+                            peer.addr,
+                        ).await {
+                            Ok(cs) => cs,
+                            Err(e) => {
+                                error!(log,
+                                    "failed to get communities for peer";
+                                    "peer" => ?peer,
+                                    "error" => ?e
+                                );
+                                return json!({
+                                    "error":
+                                        format!(
+                                            "failed to get port settings for peer {:?}: {:?}",
+                                            peer,
+                                            e
+                                        )
+                                });
+                            }
+                        };
+
+                        let allow_import = match self.datastore.allow_import_for_peer(
+                            opctx,
+                            peer.port_settings_id,
+                            &peer.interface_name,
+                            peer.addr,
+                        ).await {
```

Comment:
consider awaiting in parallel and joining.

Reply:
I've added TODO comments in this area and the next and would like to leave this as a follow-on.